clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -637,6 +638,8 @@ private void handleResponse(RequestHeader header, Struct body, long now) {
                 log.trace("Ignoring empty metadata response with correlation id {}.", header.correlationId());
                 this.metadata.failedUpdate(now);
             }
+            for (String topic : response.errorTopics(Errors.UNKNOWN_TOPIC_OR_PARTITION))
+                this.metadata.handleUnknownTopic(topic);
```

Comment:
@hachikuji I think we need to do the removal aggressively enough to avoid filling logs with messages about unknown topics. The motivation was for the REST proxy, where you could actually imagine fairly frequent metadata requests as the REST proxy instance may see the set of topics changing over time and added to on a regular basis (compared to most producer instances where the set of topics would generally be pretty much fixed).

That said, that was from a test workload that was probably unusually aggressive about adding/removing topics.